plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineSource.java

```diff
@@ -0,0 +1,226 @@
+        }
+
+        // populate
+        this.registry = matcher.group("registry");
+        this.repository = matcher.group("repository");
+        this.tag = matcher.group("tag");
+        this.digest = matcher.group("digest");
+
+    }
+
+
+    /**
+     * Build image source based on given arguments
+     * @param repository as for example codenvy/ubuntu_jdk8
+     */
+    public DockerMachineSource(String repository) {
+        super();
+        this.repository = repository;
+        setType(DOCKER_IMAGE_TYPE);
+    }
+
+    /**
+     * Defines optional tag attribute
+     * @param tag as for example latest
+     * @return current instance
+     */
+    public DockerMachineSource withTag(String tag) {
+        this.tag = tag;
+        return this;
+    }
+
+    /**
+     * Defines optional tag attribute
+     * @param tag as for example latest
+     * @return current instance
+     */
+    public void setTag(String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Defines optional registry attribute
+     * @param registry as for example docker-registry.company.com:5000
+     */
+    public void setRegistry(String registry) {
+        this.registry = registry;
+    }
+
+    /**
+     * Defines optional registry attribute
+     * @param registry as for example docker-registry.company.com:5000
+     * @return current instance
+     */
+    public DockerMachineSource withRegistry(String registry) {
+        this.registry = registry;
+        return this;
+    }
+
+    /**
+     * Defines optional digest attribute
+     * @param digest as for example sha256@1234
+     */
+    public void setDigest(String digest) {
+        this.digest = digest;
+    }
+
+    /**
+     * Defines optional digest attribute
+     * @param digest as for example sha256@1234
+     * @return current instance
+     */
+    public DockerMachineSource withDigest(String digest) {
+        this.digest = digest;
+        return this;
+    }
+
+    /**
+     * @return mandatory repository
+     */
+    public String getRepository() {
+        return this.repository;
+    }
+
+    /**
+     * @return optional tag
+     */
+    public String getTag() {
+        return this.tag;
+    }
+
+    /**
+     * @return optional registry
+     */
+    public String getRegistry() {
+        return this.registry;
+    }
+
+    /**
+     * @return optional digest
+     */
+    public String getDigest() {
+        return this.digest;
+    }
+
+    /**
+     * Returns location of this docker image, including all data that are required to reconstruct a new docker machine source.
+     */
+    public String getLocation() {
+        return getLocation(true);
+    }
+
+    /**
+     * Returns full name of docker image.
+     * <p>
+     * It consists of registry, userspace, repository name, tag, digest.
+     * E.g. docker-registry.company.com:5000/userspace1/my-repository:some-tag
```

Comment:
AFAIK it is not absolutely correct to say that this url has userspace in the given format. Docker doesn't declare userspace - only repository. Instead of that each provider of registry implement repository details as he wants. E.g. it can be absolutely correct that userspace is **/company/employee** in image **docker-registry.company.com:5000/codenvy/agaragatyi/my-repository:some-tag**